web/src/search.js

```diff
@@ -131,16 +152,7 @@ export function initialize({on_narrow_search}) {
             }
 
             if (keydown_util.is_enter_event(e) && $search_query_box.is(":focus")) {
-                // We just pressed Enter and the box had focus, which
-                // means we didn't use the typeahead at all.  In that
-                // case, we should act as though we're searching by
-                // operators.  (The reason the other actions don't call
-                // this codepath is that they first all blur the box to
-                // indicate that they've done what they need to do)
-
-                // Pill is already added during keydown event of input pills.
```

Comment:
I don't remember why I deleted this comment, but I'm also not sure what situations this actually happens, so I'll look into it. Might be able to delete it, which would be great.